### PR TITLE
Move config to env vars

### PR DESCRIPTION
**Pull request by kahowell**
I put all of the variables we currently had configured via deployment automation and a few we use regularly during development.

Configuration is still overridable via properties files, so the current `spring.config.additional-location` approach should work, until we move to just environment variables.

See README.md for documentation.

### CODE REVIEW

**Comment by kahowell** [comment]
5bfb8b1 to 491765a: s/KAFKA_API_RECONNECT_TIMEOUT/KAFKA_API_RECONNECT_TIMEOUT_MS/

File: src/main/java/org/candlepin/subscriptions/ApplicationConfiguration.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public static BeanFactoryPostProcessor servletInitializer() {
     @Bean
     public AccountListSource accountListSource(ApplicationProperties applicationProperties,
         AccountConfigRepository accountConfigRepository, ApplicationClock clock) {
-        if (applicationProperties.getAccountListResourceLocation() != null) {
+        if (!applicationProperties.getAccountListResourceLocation().isEmpty()) {
```
Comment (mstead) [inline src/main/java/org/candlepin/subscriptions/ApplicationConfiguration.java:136]: Won't this throw a NullPointerException if not set? Perhaps we should check the null case as well?

File: src/main/java/org/candlepin/subscriptions/files/ProductWhitelist.java [hunk 2]
```diff
@@ -42,7 +42,7 @@
     private final PerLineFileSource source;
 
     public ProductWhitelist(ApplicationProperties properties, ApplicationClock clock) {
-        if (properties.getProductWhitelistResourceLocation() != null) {
+        if (!properties.getProductWhitelistResourceLocation().isEmpty()) {
```
Comment (mstead) [inline src/main/java/org/candlepin/subscriptions/files/ProductWhitelist.java:45]: Won't this throw a NullPointerException if not set? Perhaps we should check the null case as well?

File: src/main/java/org/candlepin/subscriptions/ApplicationConfiguration.java (hunk 1 above)
Comment (kahowell) [reply src/main/java/org/candlepin/subscriptions/ApplicationConfiguration.java]: Fair enough, switched to `StringUtils.hasText` accordingly.

File: src/main/java/org/candlepin/subscriptions/files/ProductWhitelist.java (hunk 2 above)
Comment (kahowell) [reply src/main/java/org/candlepin/subscriptions/files/ProductWhitelist.java]: Fair enough, switched to `StringUtils.hasText` accordingly.

**Pull request merged**